packages/flutter_tools/lib/src/cache.dart

```diff
@@ -659,3 +1022,34 @@ void _ensureExists(Directory directory) {
   if (!directory.existsSync())
     directory.createSync(recursive: true);
 }
+
+class RawArtifact {
```

Comment:
Maybe call it "BinaryArtifact" instead?  It seems to be used in that sense a lot.